### PR TITLE
Adding Microsoft OpenJDK GA releases

### PR DESCRIPTION
**Pull request by d3vel0per**

### CODE REVIEW

File: bucket/microsoft11.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "description": "The Microsoft Build of OpenJDK is a new no-cost long-term supported distribution and Microsoft's new way to collaborate and contribute to the Java ecosystem.",
+    "homepage": "https://www.microsoft.com/openjdk",
+    "version": "11.0.11-9",
+    "license": {
+        "identifier": "GPL2 with Classpath Exception",
```
Comment (rasa) [inline bucket/microsoft11.json:6]:
```suggestion
        "identifier": "GPL-2.0-only WITH Classpath-exception-2.0",
```
Per https://spdx.org/licenses/GPL-2.0-with-classpath-exception.html

**Review by rasa — CHANGES_REQUESTED** [review]
This appears to be a duplicate of https://github.com/ScoopInstaller/Java/blob/master/bucket/microsoft-jdk.json . If so, please close.

**Comment by d3vel0per** [comment]
> This appears to be a duplicate of https://github.com/ScoopInstaller/Java/blob/master/bucket/microsoft-jdk.json . If so, please close.

1. Updated license identifier text
2. Delete microsoft-jdk file as:
a.  it was an absolute entry not representing an specific version
b. checkver and autoupdate entries were wrong as MS hasnt provided any endpoint to achieve the same

**Comment by se35710** [comment]
Hi,

Thank you for your contribution.

We are scraping for autoupdate. If no API is available, it's the last best thing other than manually updating manifests with every new build or release.

The checkver and hash urls were incorrect in the existing though, see below for some working snippets.

Please use this manifest for the highest numbered release. For now, that would be with `"reverse": true`.

My recommended actions are:

1. Close the other PR:s
2. Clone the manifest into microsoft16-jdk.
3. Create a new LTS manifest without `reverse`
4. Clone the LTS manifest into microsoft11-jdk.
5. Commit them all into this PR.

When JDK 17 arrives, we will review the manifest naming again.

microsoft-jdk.json:
```JSON
    "checkver": {
        "url": "https://docs.microsoft.com/java/openjdk/download",
        "regex": "(?<ms>microsoft-jdk-([\\d.]+)-windows-x64).zip",
        "reverse" : true
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://aka.ms/download-jdk/microsoft-jdk-$version-windows-x64.zip"
            }
        },
        "hash": {
            "url": "$url.sha256sum.txt"
        },
        "extract_dir": "jdk-$matchHead+$buildVersion"
    }
}
```

**Review by se35710 — CHANGES_REQUESTED** [review]
Hi,

Thank you for your contribution.

We are scraping for autoupdate. If no API is available, it's the last best thing other than manually updating manifests with every new build or release.

The checkver and hash urls were incorrect in the existing manifest though, see below for some working snippets.

Please use this manifest for the highest numbered release. For now, that would be with `"reverse": true`.

My recommended actions are:

1. Close the other PR:s
2. Clone the manifest into microsoft16-jdk.
3. Create a new LTS manifest without `reverse`
4. Clone the LTS manifest into microsoft11-jdk.
5. Commit them all into this PR.

When JDK 17 arrives, we will review the manifest naming again.

microsoft-jdk.json:
```JSON
    "checkver": {
        "url": "https://docs.microsoft.com/java/openjdk/download",
        "regex": "(?<ms>microsoft-jdk-([\\d.]+)-windows-x64).zip",
        "reverse" : true
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://aka.ms/download-jdk/microsoft-jdk-$version-windows-x64.zip"
            }
        },
        "hash": {
            "url": "$url.sha256sum.txt"
        },
        "extract_dir": "jdk-$matchHead+$buildVersion"
    }
}
```

**Comment by d3vel0per** [comment]
@se35710 I assume `reverse:true` is sorting in descending. How will it work for patches?
eg. for the highest release ie. 16.0.1 it will work as 16.0.2 > 16.0.1 when it comes
but if 11.0.11 is followed by 11.0.12. How will that be picked?

**Comment by se35710** [comment]
> [..] I assume `reverse:true` is sorting in descending. How will it work for patches?

We are assuming Microsoft will only publish the latest version. If they start to publish others, we will update the regex to include or exclude versions for a correct match.

Hopefully their JDK:s will be published by [Adoptium](https://adoptium.net), and then we can get the correct version through an API.

**Comment by d3vel0per** [comment]
@se35710 updated and merged everything in this single PR

File: bucket/microsoft-jdk.json [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
     "description": "The Microsoft Build of OpenJDK is a no-cost long-term supported distribution and Microsoft's way to collaborate and contribute to the Java ecosystem.",
     "homepage": "https://www.microsoft.com/openjdk/",
-    "version": "11.0.11.9.1",
+    "version": "16.0.1.9.1",
     "license": "GPL-2.0-only WITH Classpath-exception-2.0",
     "architecture": {
         "64bit": {
-            "url": "https://aka.ms/download-jdk/microsoft-jdk-11.0.11.9.1-windows-x64.zip",
+            "url": "https://aka.ms/download-jdk/microsoft-jdk-16.0.1.9.1-windows-x64.zip",
             "hash": "84a582be2dc3b847e07dc3b10197f74774d71537bd1ddc6b9451e487de1f5cb7"
```
Comment (se35710) [inline bucket/microsoft-jdk.json:9]: The hash should have been updated. Did you run `checkver -u -f <manifest>`?

**Review by se35710 — CHANGES_REQUESTED** [review]
1. Please double check the hash values.
2. Clone the microsoft11-jdk manifest into microsoft-lts-jdk.

**Comment by d3vel0per** [comment]
> 1. Please double check the hash values.
> 2. Clone the microsoft11-jdk manifest into microsoft-lts-jdk.

Apologies mix-up in merging the branches. Thanks for suggesting the checkver utility
Updated again as per comments

**Review by se35710 — APPROVED** [review]
LGTM

**Review by rasa — APPROVED** [review]
LGTM, but I didn't install/test locally.

**Comment by d3vel0per** [comment]
Hi @rasa, I dont have write access to merge

**Pull request merged**